test/FunctionalTests/Microsoft.OData.Core.Tests/ODataNotificationWriterTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataNotificationWriterTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.OData.Tests
+{
+    public class ODataNotificationWriterTests
+    {
+        private MemoryStream stream;
+        private TextWriter writer;
+        private IODataStreamListener streamListener;
+
+        public ODataNotificationWriterTests()
+        {
+            this.stream = new MemoryStream();
+            this.writer = new StreamWriter(this.stream);
+            this.streamListener = new MockODataStreamListener(this.writer);
+        }
+
+        [Fact]
+        public void NotificationWriterDisposeShouldInvokeStreamDisposed()
+        {
+            // We care about the notification writer being disposed
+            // We don't care about the writer passed to the notification writer
+            using (var notificationWriter = new ODataNotificationWriter(
+                new StreamWriter(new MemoryStream()),
+                this.streamListener))
+            {
+
+            }
+
+            var result = ReadStreamContents();
+
+            Assert.Equal("StreamDisposed", result);
+        }
+
+#if NETCOREAPP3_1
+        [Fact]
+        public async Task NotificationWriterDisposeShouldInvokeStreamDisposedAsync()
+        {
+            await using (var notificationWriter = new ODataNotificationWriter(
+                new StreamWriter(new MemoryStream()),
+                this.streamListener)) // `synchronous` argument becomes irrelevant

```

Comment:
Why is the synchronous argument irrelevant here? Is it because `await using` ensures that `DisposeAsync` is called?

Reply:
Yes